apps/meteor/ee/app/livechat-enterprise/server/business-hour/Helper.ts

```diff
@@ -6,9 +6,19 @@ import { LivechatBusinessHours, LivechatDepartment, LivechatDepartmentAgents, Us
 import { isEnterprise } from '../../../license/server/license';
 
 const getAllAgentIdsWithoutDepartment = async (): Promise<string[]> => {
+	// Fetch departments with agents excluding archived ones (disabled ones still can be tied to business hours)
+	// Then find the agents that are not in any of those departments
+
+	// TODO: define behaviors for disabled departmnents
+	const departmentIds = (await LivechatDepartment.findNotArchived({ projection: { _id: 1 } }).toArray()).map(({ _id }) => _id);
+
 	const agentIdsWithDepartment = (
-		await LivechatDepartmentAgents.find({ departmentEnabled: true }, { projection: { agentId: 1 } }).toArray()
+		await LivechatDepartmentAgents.find(
```

Comment:
We can make this find distinct ;)

Reply:
![image](https://github.com/RocketChat/Rocket.Chat/assets/11577696/e0b2c6e4-00ae-4f08-995d-27a96b209260)